packages/core/src/mixins/primitives/reatomArray.ts

```diff
@@ -0,0 +1,77 @@
+import { Action, Atom, action, atom } from 'src/core'
+
+export interface ArrayAtom<T> extends Atom<Array<T>> {
+  toReversed: Action<[], T[]>
+  toSorted: Action<[compareFn?: (a: T, b: T) => number], T[]>
+  toSpliced: Action<[start: number, deleteCount: number, ...items: T[]], T[]>
+  with: Action<[i: number, value: T], T[]>
+  push: Action<[...items: T[]], number>
+  pop: Action<[], T | undefined>
+  shift: Action<[], T | undefined>
+  unshift: Action<[...items: T[]], number>
+}
+
+export const reatomArray = <T>(
+  initState = [] as T[],
+  name?: string,
```

Comment:
Please, do it for all methods
```suggestion
  name: string = named('arrayAtom'),
```